src/CMakeLists.txt

```diff
@@ -536,6 +536,16 @@ zeek_target_link_libraries(zeek_objs)
 
 if (HAVE_SPICY)
     target_link_libraries(zeek_objs PRIVATE hilti spicy)
+
+    # On some platforms the transitive link of the Spicy and HILTI object
+    # libraries via e.g., `hilti` and `zeek_objs` into `zeek_exe` does not work,
+    # see #3177. Make it explicit to be sure.
+    if (BINARY_PACKAGING_MODE)
+        target_link_libraries(
+            zeek_exe PRIVATE $<IF:$<CONFIG:Debug>,hilti-rt-debug-objects,hilti-rt-objects>
+                             $<IF:$<CONFIG:Debug>,spicy-rt-debug-objects,spicy-rt-objects>)
+    endif ()
+
```

Comment:
The top-level CMakeLists.txt has this:

https://github.com/zeek/zeek/blob/6d10082cc35d9b0b4ad527cc320ff7d7ee597445/CMakeLists.txt#L956-L964

Isn't that doing the same already, but with a helper? There isn't a call to `spicy_link_object_libraries_in_tree` though. Just a wild guess.

It would be good to explain the new target_link_libraries and the quoted code - superficially they seem to do the same thing.

Reply:
Excellent point, it looks like that place place is missing the link against Spicy. I'll check whether doing that instead would fix the issue as well.

Reply:
That fixed it.